js_modules/dagster-ui/packages/ui-core/src/ticks/EvaluateScheduleDialog.tsx

```diff
@@ -64,132 +76,309 @@ export const EvaluateScheduleDialog = (props: Props) => {
-      scheduleSelector: {
-        repositoryLocationName: repoAddress.location,
-        repositoryName: repoAddress.name,
-        scheduleName: name,
+  const [selectedTimestamp, setSelectedTimestamp] = useState<{ts: number; label: string}>();
+  const scheduleSelector: ScheduleSelector = useMemo(
+    () => ({
+      repositoryLocationName: repoAddress.location,
+      repositoryName: repoAddress.name,
+      scheduleName: name,
+    }),
+    [repoAddress, name],
+  );
+
+  // query to get the schedule initially
+  const {data: getScheduleData} = useQuery<GetScheduleQuery, GetScheduleQueryVariables>(
+    GET_SCHEDULE_QUERY,
+    {
+      variables: {
+        scheduleSelector,
       },
     },
-  });
+  );
+
+  // mutation to evaluate the schedule
+  const [scheduleDryRunMutation] = useMutation<
+    ScheduleDryRunMutation,
+    ScheduleDryRunMutationVariables
+  >(SCHEDULE_DRY_RUN_MUTATION);
+
+  // mutation to launch all runs
+  const launchMultipleRunsWithTelemetry = useLaunchMultipleRunsWithTelemetry();
+
   const {
     timezone: [userTimezone],
   } = useContext(TimeContext);
   const [isTickSelectionOpen, setIsTickSelectionOpen] = useState<boolean>(false);
   const selectedTimestampRef = useRef<{ts: number; label: string} | null>(null);
   const {viewport, containerProps} = useViewport();
-  const [shouldEvaluate, setShouldEvaluate] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+  const [launching, setLaunching] = useState(false);
+
+  const [scheduleExecutionError, setScheduleExecutionError] = useState<PythonErrorFragment | null>(
+    null,
+  );
+  const [scheduleExecutionData, setScheduleExecutionData] =
+    useState<ScheduleDryRunInstigationTick | null>(null);
+
+  const canSubmitTest = useMemo(() => {
+    return getScheduleData && !submitting;
+  }, [getScheduleData, submitting]);
+
+  const submitTest = useCallback(async () => {
+    if (!canSubmitTest) {
+      return;
+    }
+    setSubmitting(true);
+
+    const repositorySelector = repoAddressToSelector(repoAddress);
+
+    const result = await scheduleDryRunMutation({
+      variables: {
+        selectorData: {
+          ...repositorySelector,
+          scheduleName: name,
+        },
+        timestamp: selectedTimestampRef.current!.ts,
+      },
+    });
+
+    const data = result.data?.scheduleDryRun;
+
+    if (data) {
+      if (data?.__typename === 'DryRunInstigationTick') {
+        if (data.evaluationResult?.error) {
+          setScheduleExecutionError(data.evaluationResult.error);
+        } else {
+          setScheduleExecutionData(data);
+        }
+      } else if (data?.__typename === 'ScheduleNotFoundError') {
+        showCustomAlert({
+          title: 'Schedule not found',
+          body: `Could not find a schedule named: ${name}`,
+        });
+      } else {
+        setScheduleExecutionError(data);
+      }
+    } else {
+      assertUnreachable('scheduleDryRun Mutation returned no data??' as never);
```

Comment:
`assertUnreachable(data);`

the point is that typescript will make data of type `never` if the conditions above successfully handled every possibility. It's purpose is to cause a typescript error if a new type is returned by the mutation to alert the dev that they need to update this callsite to handle the new return type.